### PR TITLE
Support info in federation resolve_reference

### PR DESCRIPTION
**Pull request by marcoacierno**
## Description

Add support for info for resolve_reference

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: strawberry/federation.py [hunk 1]
```diff
@@ -115,8 +116,18 @@ def entities_resolver(self, root, info, representations):
...
 
-            results.append(definition.origin.resolve_reference(**representation))
+            func_args = get_func_args(resolve_reference)
+            kwargs = representation
+
+            if "info" in func_args:
+                kwargs["info"] = info
+
+            if "root" in func_args:
+                kwargs["root"] = root
```
Comment (patrick91) [inline strawberry/federation.py:128]: is this always the value of query/mutation?

Comment (marcoacierno) [reply strawberry/federation.py:128]: I don't know how `root` works to be honest, but in the test it's what I send to `root_value`, and in the codebase where I am using this is None

Comment (patrick91) [reply strawberry/federation.py:128]: I'd remove it then, at least for now :)

Comment (marcoacierno) [reply strawberry/federation.py]: root removed

Comment (patrick91) [reply strawberry/federation.py]: root is the value of the parent of the field :)

so:

```python
@strawberry.type
class Something:
   x: int
   @strawberry.field
   def computed_x(root): # root is the same as self
       return root.x + 2
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+Add support for info param in resolve_reference
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release adds support for the info param in resolve_reference:

``````python
@strawberry.federation.type(keys=["upc"])
class Product:
    upc: str
    info: str

    @classmethod
    def resolve_reference(cls, info, upc):
        return Product(upc, info)
``````

> Note: resolver reference is used when using Federation, similar to Apollo server's `__resolveReference` https://www.apollographql.com/docs/federation/api/apollo-federation/#__resolvereference

```

File: strawberry/federation.py [hunk 3]
```diff
@@ -115,8 +116,15 @@ def entities_resolver(self, root, info, representations):
...
+            resolve_reference = definition.origin.resolve_reference
 
-            results.append(definition.origin.resolve_reference(**representation))
+            func_args = get_func_args(resolve_reference)
+            kwargs = representation
+
+            if "info" in func_args:
+                kwargs["info"] = info
+
+            results.append(resolve_reference(**kwargs))
```
Comment (patrick91) [inline strawberry/federation.py:127]: I think we'll change this in future to follow's @BryceBeagle structure, but for now this is good 😊

**Review by patrick91 — APPROVED** [review]

**Pull request merged**